lib/gollum-lib/filter/sourcemap.rb

```diff
@@ -0,0 +1,24 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Source Maps
+#
+# Find lines corresponding to h1 headers for the present markup and insert markers indicating the line number in the raw source.
+
+class Gollum::Filter::SourceMap < Gollum::Filter
+  def extract(data)
+  	regex = Gollum::Markup.formats[@markup.format][:h1]
+    return data if regex.nil?
+    i = 0
+    data.lines.map do |line|
```

Comment:
Yes, my workaround for @bartkamphorst's `#find_with_index` idea won't work when we need to map multi-line headers. :( Hopefully looping over `data.split` won't be too slow.